docs/additional/internal_aspects_en.md

```diff
@@ -0,0 +1,303 @@
+
+[dogu-integration-lib]: https://github.com/cloudogu/dogu-integration-test-library
+
+##### Trivy
+
+It is important to identify and fix the security vulnerabilities in a Docker image, or the target system.
+For this, the Trivy scanner is useful, which identifies and prioritizes security vulnerabilities in the system.
+The `dogu-build-lib` provides a [component][trivy] for this purpose, which can be used as follows to scan the image of a
+Dogu.
+
+```groovy
+EcoSystem ecoSystem = new EcoSystem(this, "gcloud-credentials-id", "ssh-credentials-id")
+Trivy trivy = new Trivy(this, ecoSystem)
+
+stage('Trivy scan') {
+    trivy.scanDogu("/dogu", TrivyScanFormat.HTML, TrivyScanLevel.CRITICAL, TrivyScanStrategy.FAIL)
+}
+```
+
+[trivy]: https://github.com/cloudogu/dogu-build-lib/blob/develop/docs/development/trivy_de.md
+
+#### Manual testing on a test stage
+
+In the manual testing phase, the Dogu should be tested in a staging environment.
+This staging environment should represent a production environment as realisitcally as possible.
+The Dogu should be installed using the `cesapp install` command.
+In addition, it should be systematically tested manually in the browser and classified as functional.
+The focus should be on the features that have the greatest relevance in production.
+
+## Definition of Done
+
+This section describes possible requirements for a final state of a development.
+It creates a common understanding for developers when a feature is done and can include various requirements.
+
+The following points describe basic requirements that can serve as the basis for a Definition of Done.
+
+### Issue tracking
+
+- An issue exists in a ticket system for every change.
+- It describes the product increment to be developed.
+- Tickets from public ticket systems are linked in internal systems.
+
+### Dockerfile
+
+Requirements for the Dockerfile can be found [here](interesting_aspects_en.md#container-images).
+
+### Dogu.json
+
+The dogu.json must contain the following configurable keys:
+- `container_config/memory_limit`
+- `container_config/swap_limit`
+
+### Dogu scripting
+
+Requirements for shell scripting can be found [here](interesting_aspects_en.md#bash-scripts).
+
+### Dogu functionalities
+
+- Consider backup & restore capability
+- Consider upgrade capability
+    - Upgrade scripts must be designed in such a way that it is possible to upgrade from all old versions of the Dogu
+      (as long as this is not prevented by the software used in the Dogu).
+    - Upgrade of Dogus from previous version must be tested.
+- Admin group
+    - Admin group must be changeable.
+    - Admin group from the `etcd` is used and if the admin group is changed the Dogu reacts accordingly
+      to it, then the old admin group will be stripped of admin rights. See e.g. SonarQube and Nexus Repository
+      manager.
+- Logging
+    - Check log output of the Dogu for errors.
+- Accepting a change of the CES FQDN
```

Comment:
CES is an informal abbreviation and should be expanded. In this case I suggest the following because we already discussed this in https://github.com/cloudogu/dogu-development-docs/blob/main/docs/important/relevant_functionalities_en.md#changeability-of-the-fqdn

```suggestion
- Accepting a change of the FQDN
```